api/features/views.py

```diff
@@ -187,6 +189,48 @@ def get_serializer_context(self):
 
         return context
 
+    def apply_state_to_queryset(
+        self, query_data: dict[str, typing.Any], queryset: QuerySet[Feature]
+    ) -> QuerySet[Feature]:
+        if not query_data.get("environment"):
+            raise serializers.ValidationError(
+                "Environment is required in order to filter by state search or by state enabled"
+            )
+        state_enabled = query_data["state_enabled"]
+        state_search = query_data["state_search"]
+        environment = query_data["environment"]
```

Comment:
Let's call this variable `environment_id`?

Reply:
Ok that's easy. However, the query_data already has it named `environment` instead of `environment_id` so that part will be kept the same for backwards compatibility.

Reply:
agreed.